test/integration/smoke/test_snapshots.py

```diff
@@ -302,14 +302,23 @@ def test_02_list_snapshots_with_removed_data_store(self):
             self.apiclient,
             vol
         )
-        self.virtual_machine_with_disk.detach_volume(
+        detach_vol_res = self.virtual_machine_with_disk.detach_volume(
             self.apiclient,
             vol
         )
 
+        if detach_vol_res.storageid != storage.id:
+            dest_store = storage
+        else:
+            all_stores = store_pools = list_storage_pools(self.apiclient)
+            for store in all_stores:
+                if store.id != storage.id:
+                    dest_store = store
```

Comment:
The dest store (to which volume is migrated) here is other than the primary storage added above. It seems, the test here is migrate the detached volume to the newly added primary storage. So, possible to attach volume to primary storage other than newly added one. Otherwise, pls update the description in the test method.

Reply:
Thanks @sureshanaparti  - you are right, I've refactored it.